website/static/js/comment.js

```diff
@@ -132,29 +146,113 @@ BaseComment.prototype.setupToolTips = function(elm) {
     });
 };
 
-BaseComment.prototype.fetch = function() {
+BaseComment.prototype.fetch = function(isCommentList, thread) {
     var self = this;
     var deferred = $.Deferred();
     if (self._loaded) {
         deferred.resolve(self.comments());
     }
+    if (thread !== undefined) {
+        return self.getThread(thread);
+    }
     $.getJSON(
         nodeApiUrl + 'comments/',
-        {target: self.id()},
+        {
+            page: self.page(),
+            target: self.id(),
+            rootId: self.rootId(),
+            isCommentList: (isCommentList || null)
+        },
         function(response) {
             self.comments(
-                ko.utils.arrayMap(response.comments.reverse(), function(comment) {
+                ko.utils.arrayMap(response.comments.reverse(), function (comment) {
                     return new CommentModel(comment, self, self.$root);
                 })
             );
+            if (isCommentList) {
+                self.discussionByFrequency(response.discussionByFrequency);
+                self.discussionByRecency(response.discussionByRecency);
+            }
             self.unreadComments(response.nUnread);
             deferred.resolve(self.comments());
+            self.checkFileExists();
             self._loaded = true;
         }
     );
     return deferred;
 };
 
+BaseComment.prototype.getThread = function(thread_id) {
+    var self = this;
+    var deferred = $.Deferred();
+    if (self._loaded) {
+        deferred.resolve(self.comments());
+    }
+    $.getJSON(
+        nodeApiUrl + 'comment/' + thread_id + '/',
+        {},
+        function(response) {
+            self.comments([]);
+            self.comments.push(new CommentModel(response.comment, self, self.$root));
+            deferred.resolve(self.comments());
+            self.checkFileExists();
+            self._loaded = true;
+        }
+    );
+    return deferred;
+}
+
+BaseComment.prototype.checkFileExists = function() {
+    var self = this;
+    var url;
+    for (var c in self.comments()) {
+        var comment = self.comments()[c];
+        if (self.level > 0 && self.loading() == false) {
+            comment.isHidden(self.isHidden());
+            if (!self.isHidden() && self.page() == 'files') {
+                comment.title(self.title());
+            }
+            comment.loading(false);
+            continue;
+        }
+        if (comment.page() !== 'files' || self.mode == 'pane') {
+            comment.loading(false);
+            continue;
+        }
+        (function(comment) {
```

Comment:
Otherwise the wrong comment would be hidden (when the ajax failure is returned the "comment" would not refer to the comment that used the url to check)